packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts

```diff
@@ -46,7 +45,11 @@ export class ToggleSemanticDateRangeFilterBar
     }
 
     async execute(): Promise<FlexCommand[]> {
-        const entitySet = isA<SmartFilterBar>(CONTROL_TYPE, this.control) ? this.control.getEntitySet() : undefined;
+        // Use a regex to match the part after the last "::" and before "--"
+        const match = this.control?.getId().match(/::([^:]+)--/);
+
+        const entitySet = match ? match[1] : undefined;
+
```

Comment:
The comment is helpful, thanks. 

Could I suggest we make this a one-liner instead?

```suggestion
        // Use a regex to match the part after the last "::" and before "--"
        const entitySet = this.control?.getId()?.match(/::([^:]+)--/)?.[1];
       
```

 It will still return undefined if something goes wrong.